android-components/components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/AddonsManagerAdapter.kt

```diff
@@ -211,14 +234,22 @@ class AddonsManagerAdapter(
     }
 
     @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
-    internal fun bindFooterButton(
-        holder: FooterViewHolder,
-    ) {
+    internal fun bindFooterButton(holder: FooterViewHolder) {
         holder.itemView.setOnClickListener {
             addonsManagerDelegate.onFindMoreAddonsButtonClicked()
         }
     }
 
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal fun bindHeaderButton(holder: HeaderViewHolder) {
+        holder.restartButton.setOnClickListener {
+            engine.enableExtensionProcessSpawning()
+            store.dispatch(ExtensionsProcessAction.ProcessEnabledAction)
+            // TODO: if we want to collect Telemetry data, we'll need to move this out of the adapter.
```

Comment:
We could use the [TelemetryMiddleware](https://github.com/mozilla-mobile/firefox-android/blob/d16cb97beb0443f78ac0a30a8455d2a7d6b69f6e/fenix/app/src/main/java/org/mozilla/fenix/telemetry/TelemetryMiddleware.kt#L46) by adding a new case for this action   :)

Reply:
Good idea, thanks